session.go

```diff
@@ -700,6 +706,30 @@ func (s *Session) deallocateHandle(l *link) {
 	s.handles.Remove(l.handle)
 }
 
+func (s *Session) abandonLink(l *link) {
+	s.abandonedLinksMu.Lock()
+	defer s.abandonedLinksMu.Unlock()
+	s.abandonedLinks = append(s.abandonedLinks, l)
+}
+
+func (s *Session) freeAbandonedLinks() error {
+	s.abandonedLinksMu.Lock()
+	defer s.abandonedLinksMu.Unlock()
+
+	for _, l := range s.abandonedLinks {
```

Comment:
It'd be interesting to log how many abandoned links are here.

Reply:
Added logging for this and abandoned sessions.